### ISSUE TITLE
[RTE]: known issues with RTE (after its update)

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#1422]
## Description
All these issues regarding RTE after updating it on [Plate](https://plate.udecode.io/docs/playground).

## Table

1. If User inserts a table, a blank line is inserted before it. User can only delete it with the `Backspace` key. 
In the video, the `Delete` key is first pressed, after which the cursor will move to the first line of the table.

https://github.com/epam/UUI/assets/57751283/c992a694-1972-4088-89af-5b758ffab4b8

2. There is **no possibility** to add a new empty line after the inserted table.

3. When merging cells in a row, the cursor moves to the cell on the row above.

https://github.com/epam/UUI/assets/57751283/8e036ccb-f693-499d-8ed8-3844faa37f45

4. After vertically merging table cells, other cells are automatically selected. With horizontal merging - everything is ok. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)** 

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/0f0f64eb-cd3c-475d-8f1f-a74351d584ae

5. It is possible to improve the removal of columns when there are merged cells. Compress or split merged. Depending on the case. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

https://github.com/epam/UUI/assets/57751283/6bc5dec1-4234-4485-a781-eb85561eb74c

6. If you insert a table in _**Safari**_ immediately after the text using cmd + v, then only the text is inserted. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/1feba8a6-afc2-4b8d-b7f1-0d68d905c14c

7. Add the ability to split cells vertically. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

8. Resizing borders are still not ideal in edge cases. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**
![image](https://github.com/epam/UUI/assets/57751283/4c21e340-035c-441e-9d84-5914d1115c86)

9. In _**Firefox**_, User cannot select more than 2 horizontal cells at once. In general, in Plate, selection in the table does not work properly. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/f6d5ca21-6b96-4463-8895-e6175670a377

10. In _**Firefox**_ the borders have gaps

**_not actual (qa)_**

<img width="783" alt="Screenshot 2024-02-26 at 11 07 26" src="https://github.com/epam/UUI/assets/57751283/e32f3062-5cb0-4b74-b574-71bbd4b615b3">

11. When copying a table from Word to RTE, the header style is not copied, i.e. header style is reset to normal. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**
![image](https://github.com/epam/UUI/assets/57751283/1b15c4f0-2ecd-4cf7-982f-44b75f40a238)

12. Selecting cells after the merge does not work well. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

**_not actual (qa)_**
![image](https://github.com/epam/UUI/assets/57751283/c4a524d7-dd20-402b-bd3c-33d5292e2254)

13. After splitting a merged cell, re-merging does not work correctly. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/653bd273-28e5-4fc7-8c47-35be1bb99fd1

14. When merging *header* cells, the background and font of the merged cell become the same as a normal cell. This is a problem on the slate side

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/bbd3fda5-fdcc-4da5-9602-5b3b645444f8

## Text (+copy/paste)

1. If User copies a text with a picture from the Word, then the picture is not inserted. The picture can be inserted when copying only pictures without text.

https://github.com/epam/UUI/assets/57751283/0cf09ea0-e22c-42c9-ad49-4b78d07a2d8c

2. Problems in _Firefox_ with deselecting bold, italic, underline, and superscript for one word. If User tries to highlight a single word by double-clicking on the left mouse button, then the word + space after it is automatically selected, in which case the bold font is not canceled. But if User selects strictly one word with the mouse (without a space), then the word becomes a regular font.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/d9b5ee2e-8e85-41ca-ba40-c9a69027de8e

3. Headers not copied from Word.

![image](https://github.com/epam/UUI/assets/57751283/3c9f771e-f5d8-4be9-983c-b49dcda66eea)

4. List type change does not work when switching from checkboxes. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

https://github.com/epam/UUI/assets/57751283/277538cf-5ed7-4880-8f06-3b6afd514b7f

5. Lists copied from Word don't copy well. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

**_Fixed (qa)_**

https://github.com/epam/UUI/assets/57751283/514f6530-cfe8-4110-83f4-05962ff5c02a

https://github.com/epam/UUI/assets/57751283/6ad615aa-9b74-40f9-865f-bfadd1be3e48

6. Selecting all text and trying to delete it does not remove the splitter. If all lists are deleted, when all text is selected, the splitter starts to be deleted. **It seems that this issue is related to the Plate.js library, let's try to make this fix on Plate.js side (create issue or MR)**

**_Fixed_**

https://github.com/epam/UUI/assets/57751283/283ce6ac-9aed-4d41-8de4-f57d3f665daa

## General

1. Sometimes even when the focus is not in the editor, the element inside the area is still highlighted. So usually only when there is only one element.
![image](https://github.com/epam/UUI/assets/57751283/1d82384c-f8c1-427c-9776-6224e981a21a)
![image](https://github.com/epam/UUI/assets/57751283/5b1ddfc5-7a87-45ed-b31f-0056044da92d)

2. In an empty RTE in **_Safari_**, there is an empty line before the placeholder, which is not removed. When focus moves away from the editor, the placeholder is outside of the RTE.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/03eddfc8-fa41-4844-b274-6ddc70cc3561

3. The `Delete` button does not delete all items in Safari

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/0bd0cafd-fb6a-4ff2-8a2d-abe3f33cd77d

4. "Add description" does not disappear when a "Block" is added to an empty RTE. It disappears as soon as the user starts typing text. But it should disappear as soon as a "Block" is added to an empty RTE.

**_not actual (qa)_**

<img width="957" alt="Screenshot 2023-08-09 at 14 03 09" src="https://github.com/epam/UUI/assets/57751283/36fb2b8d-7dc3-4fff-b963-528ad585a701">

5. In **Firefox** sometimes text can be edited only after second click.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/4cfe1411-4278-47a3-9288-bf5d79d88a9f

6. When inserting an image in the Rich Text Editor using the modal window, the cursor focus is not restored to its previous position. Instead, it jumps to the top of the editor field. The cursor should remain in its current position without jumping after closing the modal.

https://github.com/user-attachments/assets/6314fe56-d84e-4964-a802-5930433ba7bf

## Lists

1. If User makes a list with checkboxes in the empty editor, she/he cannot delete the line with the backspace key. In the video, the backspace key is pressed.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/d9cb8773-0923-4a2d-af4d-5237f32c00ee

2. If we have a list with checkboxes, and in the line checkbox is selected, a new line after is also created with the selected checkbox.

https://github.com/epam/UUI/assets/57751283/d544de78-8caf-42ee-b419-5fe43f625deb

3. In lists with a checkbox, pressing the Enter key again on the last line does not end the list, leaving an empty line with the checkbox.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/29a97cde-c35e-4c5e-a39a-3a7c5d1cbe7c

4. Several separate lists with checkboxes are not merged into one.

https://github.com/epam/UUI/assets/57751283/e9039e4d-9de3-4699-ad88-9ab2e51ec1d8

5. Lists from bullets and numbers do not change to checkboxes.

https://github.com/epam/UUI/assets/57751283/9195f49f-e355-4fa3-b6f0-1d842a648e9b

6. Tables -> lists with checkboxes -> checkbox can be selected, but toolbars disappear after a couple of selects. Possible reason: toolbars disappear due to the fact that on the checkbox check the selection jumps inside the checkbox content.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/16c5e982-889c-41ef-91e3-f28378bcaba3

## Media

1. When drag-and-drop files in _Firefox_, insertion point cursor is not visible.

**_not actual (qa)_**

![image](https://github.com/epam/UUI/assets/57751283/c5a5c5d7-e581-4c1a-b964-64b0d7c73dbb)
![image](https://github.com/epam/UUI/assets/57751283/09303f62-9905-4e29-9b6e-67f25e6806ea)

2. If the cursor is activated one line above a previously added file, the file is added below the previously added file.

https://github.com/epam/UUI/assets/57751283/87738065-952b-44a1-bcb8-f14aada36e4a

3. Сursor changes to pointer in view mode when hovered over pictures

![image](https://github.com/epam/UUI/assets/129292145/bedb8299-ec52-4cd1-9fed-365b0877544b)

## Blocks
1. If User selects text with a double left mouse click, the entire text of the block is copied.

**_not actual (qa)_**

https://github.com/epam/UUI/assets/57751283/c7a08e9f-ad1d-450b-a5de-323efe583498

## Pictures
1. There is no option to remove a blank line between two pictures. When User tries to delete a line, the top image is deleted.

https://github.com/epam/UUI/assets/57751283/be46800f-0395-4635-8938-69dba1cc1807

**Comment by AlekseyManetov** [comment]
Case 17 was fixed by #1530
